ui/src/slider.tsx

```diff
@@ -63,11 +63,19 @@ export const
     const
       { min = 0, max = 100, step = 1, value = 0 } = m,
       defaultValue = (value < min) ? min : ((value > max) ? max : value),
-      onChange = (v: U) => wave.args[m.name] = v,
+      [val, setVal] = React.useState(defaultValue),
+      onChange = (v: U) => {
+        wave.args[m.name] = v
+        m.value = v
+        setVal(v)
+      },
       onChanged = React.useCallback((_e: MouseEvent | KeyboardEvent | TouchEvent, _value: U) => { if (m.trigger) wave.push() }, [m.trigger])
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    React.useEffect(() => { wave.args[m.name] = defaultValue }, [])
+    React.useEffect(() => {
+      wave.args[m.name] = defaultValue
+      setVal(defaultValue)
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [value])
```

Comment:
Shouldn't there be `defaultValue` as a dependency rather than `value`? Also, let's add `name` dep.